src/linux/DisplayInfoLinux.cpp

```diff
@@ -10,16 +14,36 @@ using namespace VSTGUI;
     
 float getDisplayBackingScaleFactor(CFrame *)
 {
-    Surge::UserInteractions::promptError("getDisplayBackingScaleFactor not implemented yet on linux.",
-                                         "Software Error");
+    std::cerr << "BackingDisplayFactor is not implemented on linux. Returning 1.0" << std::endl;
     return 1.0;
 }
+
+
+/*
+** We make a static single screen assumption for now so cache the screen size and only
+** query Xlib for it once.
+*/    
+int dispinfoScreenW = -1;
+int dispinfoScreenH = -1;
     
 CRect getScreenDimensions(CFrame *)
 {
-    Surge::UserInteractions::promptError("getScreenDimensions not implemented yet on linux. Returning 1400x1050",
-                                         "Software Error");
-    return CRect(CPoint(0,0), CPoint(1400,1050)); // a typical mid-range 15" laptop
+    if(dispinfoScreenW < 0 || dispinfoScreenH < 0 )
+    {
+        /*
+        ** For now make the assumption of single screen to at least avoid
+        ** clip and bound errors like shown in #594
+        **
+        ** FIXME: Deal with multi-display linux
+        */
+        Display* d = XOpenDisplay(NULL);
```

Comment:
looks good, though I would add a verification here on the `d` pointer for nullness.
Also, since we are using C++14/17 after all, no need to use NULL, use nullptr instead :)
Copy&pasted from somewhere I guess :P